frontend-build-pipeline.yml

```diff
@@ -16,27 +19,48 @@ parameters:
 - name: sonarCloudProjectName # sonar cloud's cli project name
   type: string
   default: 'RevatureShopFrontend'
+- name: sonarCloudExclude # files for sonar cloud to ignore
+  type: string
+  default: '**/*.spec.ts'
+- name: sonarCloudCoverageDir
+  type: string
+  default: '$(Build.Repository.LocalPath)/coverage/lcov.info'
+- name: sonarCloudSources
+  type: string
+  default: '.'
 
 steps:
 
-
-
 # Node.js is required to build Angular apps
 - task: NodeTool@0
   inputs:
     versionSpec: '12.14'
   displayName: 'Install Node.js'
 
 - script: |
-    npm ci
+    ls $(Build.Repository.LocalPath)
+
+- script: |
+    npm install
```

Comment:
re. https://github.com/1054-Daniel-Felleman-Revature-Swag-Shop/RevatureShopFrontend/pull/1#discussion_r676759574

you run npm install when you intentionally want to update the package-lock file, such as when you've edited the package.json dependencies. so run that manually and commit the changes it makes to the package lock. then let the pipeline continue to do npm ci.
i see what you're saying about pipeline robustness, but that negates the point of the lock file. the lock file is there so that the app doesn't suddenly break because the latest version of some nested dependency (which an npm install with an empty node_modules is likely to grab) has a critical bug. it really is just up to whoever modifies the package.json to ensure the lockfile is also updated by running npm install.

Reply:
Ah okay. Updated package-lock.json and switched back to using npm ci.